app/src/features/apiClient/helpers/APIClientManager/modules/scripts/workerScripts.ts

```diff
@@ -0,0 +1,170 @@
+/* eslint-disable no-new-func */
+
+import { ScriptExecutedPayload } from "./types";
+
+export const requestWorkerFunction = function (e: MessageEvent) {
+  const { script, request, currentVariables } = e.data;
+
+  const mutations: ScriptExecutedPayload["mutations"] = {
+    environment: {
+      $set: {},
+      $unset: {},
+    },
+  };
+
+  const logDummyErrorMessage = () => {
+    console.log("This Method is not supported.");
+    return undefined;
+  };
+
+  const sandbox = {
+    rq: {
+      request,
+      environment: {
+        set: (key: string, value: any) => {
+          mutations.environment.$set[key] = value;
+        },
+        get: (key: string) => {
+          const variable = currentVariables[key];
+          return variable?.localValue || variable?.syncValue;
+        },
+        remove: (key: string) => {
+          mutations.environment.$unset[key] = "";
+        },
+      },
+      variables: new Proxy(
+        {},
+        {
+          get: () => logDummyErrorMessage,
+        }
+      ),
```

Comment:
This might fail in case when the property which is accessed is not a function. 
eg. `rq.environment.name`